packages/grid/x-data-grid/src/hooks/features/selection/useGridSelection.ts

```diff
@@ -388,25 +395,70 @@ export const useGridSelection = (
         return;
       }
 
-      // Get the most recent params because the cell mode may have changed by another listener
-      const cellParams = apiRef.current.getCellParams(params.id, params.field);
-      const isEditMode = cellParams.cellMode === GridCellModes.Edit;
-      if (isEditMode) {
+      // Get the most recent cell mode because it may have been changed by another listener
+      if (apiRef.current.getCellMode(params.id, params.field) === GridCellModes.Edit) {
         return;
       }
 
       if (event.key === ' ' && event.shiftKey) {
         event.preventDefault();
-        handleSingleRowSelection(cellParams.id, event);
+        handleSingleRowSelection(params.id, event);
         return;
       }
 
       if (event.key.toLowerCase() === 'a' && (event.ctrlKey || event.metaKey)) {
         event.preventDefault();
         selectRows(apiRef.current.getAllRowIds(), true);
       }
+
+      if (isNavigationKey(event.key) && event.shiftKey) {
```

Comment:
Could you create an issue or explain in the PR description what you're fixing here? https://github.com/mui/mui-x/pull/4157/#issuecomment-1065146849

Reply:
https://github.com/mui/mui-x/issues/4185

Here it is

Reply:
Yeah, it's a bug. Can you move this code to another PR?

Reply:
Selection with <kbd>Shift</kbd> + arrow keys is still broken. I would keep this PR as small as possible, only fixing #4156, which is simple.

Reply:
What I said is that I am doing the opposite
Instead of moving 90% of this PR to another one, I am moving the 10% related to #4156 and keeping the rest here.

It does not change anything, it's just easier to do.

Reply:
https://github.com/mui/mui-x/pull/4196